explanation/dnssec/dnssec.md

```diff
@@ -0,0 +1,254 @@
+
+## What is it?
+
+DNSSEC stands for Domain Name System (DNS) Security Extensions (SEC). It's an extension to DNS that introduces digital signatures so that each DNS response can be verified for:
+
+ * integrity: The answer was not tampered with and did not change during transit.
+ * authenticity: The data came from the true source, and not another entity impersonating the source.
+
+It's important to note that DNSSEC, however, will NOT encrypt the data: it is still sent in the clear.
+
+DNSSEC is based on public key cryptography, meaning that every DNS zone has a public/private key pair. The private key is used to sign the zone's DNS records, and the corresponding public key can be used to verify those signatures. This public key is also published in the zone, and anyone fetching records can also fetch the public key to verify the signature of the data.
+
+The question then becomes, how to trust that this public key is authentic? Turns out the key is also signed: it's signed by the parent zone's key, which is also signed by its parent, and so on, all the way to the top: the root DNS zone. The root zone keys are trusted implicitly, and all DNS resolvers have it as a trust anchor. This sequence of keys and signatures all the way to the top is called the chain of trust.
+
+The public key cryptography behind SSL/TLS is similar, but there we have the Certificate Authority entity (CA) that issues the certificates and vouches for them. Every single web browser or other SSL/TLS client or operating system needs to have a "bootstrap" list of CAs that it will trust by default, and there are dozens. In DNSSEC, the only bootstrap public key the resolver needs is the root zone one. It's as if there were only one trusted CA.
+
+For a more detailed explanation of how the DNSSEC validation is performed, please refer to the [Simplified 12-step DNSSEC validation process](https://bind9.readthedocs.io/en/latest/dnssec-guide.html#the-12-step-dnssec-validation-process-simplified) guide from ISC.
+
+## New resource records (RRs)
+DNSSEC introduces a set of new Resource Records. Here are the most important ones:
+
+ * RRSIG: Resource Record Signature. Each RRSIG record matches a corresponding Resource Record, i.e., it's the digital cryptographic signature of that Resource Record.
+ * DNSKEY: There are several types of keys used in DNSSEC, and this record is used to store the public key in each case.
+ * DS: Delegation Signer. This stores a secure delegation, and is used to build the authentication chain to child DNS zones. This makes it possible for a parent zone to "vouch" for its child zone.
+ * NSEC, NSEC3, NSEC3PARAM: Next Secure record. These records are used to prove that a DNS name does not exist.
+
+As an example, when a query is performed by a client that declares it understands DNSSEC, and the resource record it's querying for has a signature, the RRSIG is also returned:
```

Comment:
"For instance, when a DNSSEC-aware client queries a resource record that is signed, the corresponding RRSIG record is also returned."

Reply:
much better